Enable WebKit process extensions on newer OS versions

pvollan opened the pull request:
#### 61af27d48c14dc47d1fe5592ebf16936da45542b
<pre>
Enable WebKit process extensions on newer OS versions
<a href="https://bugs.webkit.org/show_bug.cgi?id=270444">https://bugs.webkit.org/show_bug.cgi?id=270444</a>
<a href="https://rdar.apple.com/124716012">rdar://124716012</a>

Reviewed by Chris Dumez.

This was previously landed in &lt;<a href="https://commits.webkit.org/275857@main">https://commits.webkit.org/275857@main</a>&gt;.

* Source/WTF/wtf/PlatformUse.h:
* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/BaseXPCService.xcconfig:
* Source/WebKit/Configurations/GPUService.xcconfig:
* Source/WebKit/Configurations/NetworkService.xcconfig:
* Source/WebKit/Configurations/WebContentService.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:

Canonical link: <a href="https://commits.webkit.org/277078@main">https://commits.webkit.org/277078@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cd2f08ea8c2abc011d734fc2ab49d86df0c8d752

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44963 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/24072 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47462 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47621 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40970 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/28182 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21470 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36912 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45541 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21165 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38721 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17998 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18587 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39852 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/3011 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/38163 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/41271 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40154 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49293 "Built successfully") | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/44420 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/repro_1289.js.wasm-omg (failure)") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19935 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16483 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43909 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/21251 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42678 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21605 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/51591 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6517 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20930 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10478 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (commented):
Seems reasonable. What was wrong with the original commit, and how is this different?

pvollan committed:
Enable WebKit process extensions on newer OS versions
https://bugs.webkit.org/show_bug.cgi?id=270444
rdar://124716012

Reviewed by Chris Dumez.

This was previously landed in <https://commits.webkit.org/275857@main>.

* Source/WTF/wtf/PlatformUse.h:
* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/BaseXPCService.xcconfig:
* Source/WebKit/Configurations/GPUService.xcconfig:
* Source/WebKit/Configurations/NetworkService.xcconfig:
* Source/WebKit/Configurations/WebContentService.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:

Canonical link: https://commits.webkit.org/277078@main
